app/main.py

```diff
@@ -1,2 +1,52 @@
+from typing import Any, Hashable, List
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table: List[List[Node]] = [[] for _ in range(self.capacity)]
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [[] for _ in range(self.capacity)]
+        self.length = 0
```

Comment:
consider reducing length by 1 when an item is replaced. For example, if self[key] = value is called for an existing key, length should stay the same.